engines/scumm/imuse_digi/dimuse_internalmixer.cpp

```diff
@@ -0,0 +1,1105 @@
+	if (feedSize == inFrameCount) {
+		if (inFrameCount / 2) {
+			srcBuf_ptr = srcBuf;
+			for (int i = 0; i < inFrameCount / 2; i++) {
+				mixBufCurCell[0] += *((uint16 *)ampTable + (srcBuf_ptr[0] | ((srcBuf_ptr[1] & 0xF)  << 8)));
+				mixBufCurCell[1] += *((uint16 *)ampTable + (srcBuf_ptr[2] | ((srcBuf_ptr[1] & 0xF0) << 4)));
+				srcBuf_ptr += 3;
+				mixBufCurCell += 2;
+			}
+		}
+	} else if (2 * inFrameCount == feedSize) {
+		srcBuf_ptr = srcBuf;
+		if ((inFrameCount / 2) - 1) {
+			for (int i = 0; i < (inFrameCount / 2) - 1; i++) {
+				value = *((uint16 *)ampTable + (srcBuf_ptr[2] | ((srcBuf_ptr[1] & 0xF0) << 4)));
+
+				mixBufCurCell[0] += *((uint16 *)ampTable + (srcBuf_ptr[0] | ((srcBuf_ptr[1] & 0xF) << 8)));
+				mixBufCurCell[1] += (value + *((uint16 *)ampTable + value)) >> 1;
+				mixBufCurCell[2] += value;
+				mixBufCurCell[3] += ((int16)value + *((int16 *)ampTable + (srcBuf_ptr[3] | ((srcBuf_ptr[4] & 0xF) << 8)))) >> 1;
+				srcBuf_ptr += 3;
+				mixBufCurCell += 4;
+			}
+		}
+		mixBufCurCell[0] +=  *((uint16 *)ampTable + (srcBuf_ptr[0] | ((srcBuf_ptr[1] & 0xF)  << 8)));
+		mixBufCurCell[1] += (*((uint16 *)ampTable + (srcBuf_ptr[0] | ((srcBuf_ptr[1] & 0xF)  << 8)))
+			               + *((uint16 *)ampTable + (srcBuf_ptr[2] | ((srcBuf_ptr[1] & 0xF0) << 4)))) >> 1;
+		mixBufCurCell[2] +=  *((uint16 *)ampTable + (srcBuf_ptr[2] | ((srcBuf_ptr[1] & 0xF0) << 4)));
+		mixBufCurCell[3] +=  *((uint16 *)ampTable + (srcBuf_ptr[2] | ((srcBuf_ptr[1] & 0xF0) << 4)));
+	} else if (2 * feedSize == inFrameCount) {
+		if (feedSize) {
+			srcBuf_ptr = srcBuf;
+			for (int i = 0; i < feedSize; i++) {
+				mixBufCurCell[i] += *((uint16 *)ampTable + (srcBuf_ptr[0] | ((srcBuf_ptr[1] & 0xF) << 8)));
+				srcBuf_ptr += 3;
+			}
+		}
+	} else {
+		xorFlag = 0;
+		residualLength = -inFrameCount;
+		if (feedSize) {
+			srcBuf_ptr = srcBuf;
+			for (int i = 0; i < feedSize; i++) {
+				if (xorFlag) {
+					term_1 = (srcBuf_ptr[1] & 0xF0) << 4;
+					term_2 = srcBuf_ptr[2];
+				} else {
+					term_1 = (srcBuf_ptr[1] & 0xF) << 8;
+					term_2 = srcBuf_ptr[0];
+				}
+
+				mixBufCurCell[i] += *((uint16 *)ampTable + (term_2 | term_1));
+				residualLength += inFrameCount;
+				while (residualLength >= 0) {
+					residualLength -= feedSize;
+					xorFlag ^= 1u;
+					if (!xorFlag)
+						srcBuf_ptr += 3;
+				}
+			}
+		}
+	}
+}
+
+void IMuseDigiInternalMixer::mixBits16Mono(uint8 *srcBuf, int inFrameCount, int feedSize, int mixBufStartIndex, int *ampTable) {
+	uint16 *mixBufCurCell;
+	uint16 *srcBuf_ptr;
+	int residualLength;
+
+	mixBufCurCell = (uint16 *)(&_mixBuf[2 * mixBufStartIndex]);
+	if (feedSize == inFrameCount) {
+		if (feedSize) {
+			srcBuf_ptr = (uint16 *)srcBuf;
+			for (int i = 0; i < feedSize; i++) {
+				mixBufCurCell[i] += *(uint16 *)((uint8 *)ampTable + (((int16)srcBuf_ptr[i] & (int16)0xFFF7) >> 3) + 4096);
+			}
+		}
+	} else if (2 * inFrameCount == feedSize) {
+		srcBuf_ptr = (uint16 *)srcBuf;
+		int i = 0; // TODO: check
```

Comment:
what to check here? is actual TODO?

Reply:
Whoops, this is a leftover from august, probably, thanks!